misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -464,6 +472,19 @@ public void sendOutput(String output, String category) {
         }
     }
 
+    public Value evaluateExpression(String expression, ThreadReference threadReference) {
+        try {
+            StackFrameProxyImpl frame = getAllThreads().get(threadReference.uniqueID()).frame(0);
+            SuspendedContext ctx = new SuspendedContext(project, context.getDebuggeeVM(),
+                    getAllThreads().get(threadReference.uniqueID()), frame);
+            ExpressionEvaluator evaluator = new ExpressionEvaluator(ctx);
+            return evaluator.evaluate(expression);
+        } catch (JdiProxyException e) {
+            LOGGER.error(e.getMessage(), e);
+            return null;
+        }
+    }
+
```

Comment:
Shall we make this package-private + static?

Reply:
Cannot make to `static` as this method is using non-static members. This method is written in `JBallerinaDebugServer` class to get the maximum use of its class variable and used in `JDIEventProcessor` class. Therefore it cannot be made `private`. Therefore keeping it as `public`

Reply:
@praveennadarajah I meant, `package-private` :slightly_smiling_face: Also, whats the reason for keeping this util inside `JballerinaDebugServer`, since this is used only within `JDIEventProcessor`? (If your intention is to use the state variables of `JBallerinaDebugServer`, those are already accessible from the `JDIEventProcessor`)

Reply:
Moved the util method to `JDIEventProcess` class and made it private.